lib/Sema/TypeCheckDeclOverride.cpp

```diff
@@ -1524,8 +1524,10 @@ static bool checkSingleOverride(ValueDecl *override, ValueDecl *base) {
   if (auto baseVar = dyn_cast<VarDecl>(base)) {
     if (auto classDecl =
           baseVar->getDeclContext()->getAsClassOrClassExtensionContext()) {
-      if (classDecl->getBaseName().userFacingName() == "NSObject" &&
-          baseVar->getBaseName().userFacingName() == "hashValue") {
+      if (baseVar->getName() == ctx.Id_hashValue &&
+          classDecl->getName().is("NSObject") &&
+          (classDecl->getModuleContext()->getName().is("ObjectiveC") ||
+           classDecl->getModuleContext()->getName().is("Foundation"))) {
```

Comment:
I believe these two are also KnownIdentifiers. (We're not hugely consistent about what we use those for. Maybe it's less important for checking anyway.)

Reply:
So they are!